scalyr_agent/log_processing.py

```diff
@@ -183,7 +221,14 @@ def __init__(self, path, config, log_config, file_system=None, checkpoint=None):
         self.__log_deletion_delay = config.log_deletion_delay  # Defaults to 10 * 60
         self.__page_size = config.read_page_size  # Defaults to 64 * 1024
 
-        self.__parse_as_json = log_config.get('parse_lines_as_json', False)
+        self.__parse_format = log_config.get( 'parse_format' )
+        parse_as_json = log_config.get('parse_lines_as_json', None, none_if_missing=True)
```

Comment:
> 2. But in k8s and docker monitors, __create_log_config seems to be creating dicts.

Which are then converted to JsonObjects later on when `parse_log_config` is called (see e.g. in `copying_manager.py`, `def add_log_config`

So it's definitely expecting a JsonObject, it's just until the `parse_lines_as_json` option was deprecated, we never needed to use JsonObject specific code (the `none_if_missing` param), and so all the tests that were incorrectly passing in `dicts` didn't have a problem.

I've made it so the code will handle either.  Probably it would have been better to make the tests pass in JsonObjects but there were a lot of cases doing this, and it's not clear if there is anywhere else doing this either.